tests/test_grid3d/test_opm_integration.py

```diff
@@ -0,0 +1,116 @@
+import os
+
+import pytest
+from hypothesis import HealthCheck, given, settings
+from numpy.testing import assert_allclose
+
+import xtgeo
+
+from .grid_generator import xtgeo_grids
+
+
+def deck_contents(dimensions, load_statement):
+    nx, ny, nz = dimensions
+    size = nx * ny * nz
+    return f"""
+RUNSPEC
+DIMENS
+ {nx} {ny} {nz} /
+OIL
+WATER
+NOSIM
+WELLDIMS
+ 0 /
+EQLDIMS
+/
+TABDIMS
+   1* 1* 30 30 /
+GRID
+{load_statement}
+PORO
+ {size}*0.1 /
+PERMX
+ {size}*150 /
+PERMY
+ {size}*150 /
+PERMZ
+ {size}*15 /
+PROPS
+SWOF
+ 0.0 0.0  1.0 0.0
+ 1.0 1.0  0   0.0 /
+DENSITY
+ 3*40 /
+PVTW
+ 1000 1.0 0.0 0.0 0.0 /
+PVTO
+  0.0 500.0  1.0 0.1
+      5000.0 0.0 0.2  /
+  3.0 1500.0 2.0 0.1
+      5000.0 0.0 0.2  /
+/
+SOLUTION
+EQUIL
+       3*1000 0 1000 0 0 0 0 /
+SUMMARY
+SCHEDULE
+NOSIM
+END
+"""
+
+
+def convert_to_egrid(tmpdir, dimensions, grdecl):
+    deck = tmpdir / "test.DATA"
+    load_statement = f"INCLUDE\n '{grdecl}' /"
+    with tmpdir.as_cwd():
+        with open(deck, "w") as fh:
+            fh.write(deck_contents(dimensions, load_statement))
+        os.system(f"flow {deck}")
+    return tmpdir / "TEST.EGRID"
+
+
+def read_write_egrid(tmpdir, dimensions, egrid):
```

Comment:
There is a `setup_tmpdir` fixture in `conftest.py` that will abstract away moving to `tmpdir`

Reply:
Done!